src/constraints.jl

```diff
@@ -239,3 +239,98 @@ function set_coefficient(constraint::ConstraintRef{Model, MOICON{F, S}},
         MOI.ScalarCoefficientChange(index(variable), convert(T, value)))
     return
 end
+
+"""
+    shadow_price(constraint::ConstraintRef)
+
+The change in the objective from an infinitesimal relaxation of the constraint.
+This value is computed from [`result_dual`](@ref) and can be queried only when
+`has_result_dual` is `true` and the objective sense is `MinSense` or `MaxSense`.
+For linear constraints, the shadow prices differ at most in sign from the
+`result_dual` value depending on the objective sense.
+
+## Notes
+
+- The function simply translates signs from `result_dual` and does not validate
+  the conditions needed to guarantee the sensitivity interpretation of the
+  shadow price. The caller is responsible, e.g., for checking whether the solver
+  converged to an optimal primal-dual pair or a proof of infeasibility.
+- The computation is based on the current objective sense of the model. If this
+  has changed since the last solve, the results will be incorrect.
+- Relaxation of equality constraints (and hence the shadow price) is defined
+  based on which sense of the equality constraint is active.
+"""
+function shadow_price(constraint::ConstraintRef{Model, MOICON})
+    error("The shadow price is not defined or not implemented for this type " *
+          "of constraint.")
+end
+
+# Internal function.
+function shadow_price_less_than_(dual_value, sense::MOI.OptimizationSense)
+    # When minimizing, the shadow price is nonpositive and when maximizing the
+    # shadow price is nonnegative (because relaxing a constraint can only
+    # improve the objective). By MOI convention, a feasible dual on a LessThan
+    # set is nonpositive, so we flip the sign when maximizing.
+    if sense == MOI.MaxSense
```

Comment:
Why not merge the two functions into one and do `sense == MOI.MaxSense ⊻ greater_than` ?
One might argue that it decreases readability but the fact that the comment in the other function references this one means that merging them might also increase readability.

Reply:
I understand the two cases separately. Also both functions are small enough to fit on a single screen together, so readability seems fine to me.